src/planner/binder.cpp

```diff
@@ -24,6 +23,14 @@ Binder::Binder(ClientContext &context, Binder *parent_)
 	}
 }
 
+void Binder::DisableParentCTEs() {
+	unordered_map<string, shared_ptr<Binding>> cleared_bindings;
```

Comment:
Rather than having a function "DisableParentCTEs", I think it would be cleaner to add a flag to the Binder constructor that never inherits them in the first place (that defaults to true). E.g. `Binder(ClientContext &context, Binder *parent, bool inherit_ctes=true);` Right now first the CTEs of the parent are copied, and afterwards they are stripped again.